include/nbl/builtin/hlsl/emulated/emulated_float64_t_impl.hlsl

```diff
@@ -0,0 +1,327 @@
+#ifndef _NBL_BUILTIN_HLSL_EMULATED_FLOAT64_T_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_EMULATED_FLOAT64_T_IMPL_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/ieee754/ieee754.hlsl>
+#include <nbl/builtin/hlsl/algorithm.hlsl>
+#include <nbl/builtin/hlsl/tgmath.hlsl>
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+
+// TODO: when it will be possible, use this unions wherever they fit:
+/*
+* union Mantissa
+* {
+*   struct
+*   {
+*       uint32_t highBits;
+*       uint64_t lowBits;
+*   };
+*
+*   uint32_t2 packed;
+* };
+*
+*/
+/*
+* union Mantissa
+* {
+*   struct
+*   {
+*       uint64_t lhs;
+*       uint64_t rhs;
+*   };
+*
+*   uint32_t4 packed;
+* };
+*
+*/
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace impl
+{
+NBL_CONSTEXPR_INLINE_FUNC uint64_t2 shiftMantissaLeftBy53(uint64_t mantissa64)
+{
+    uint64_t2 output;
+    output.x = mantissa64 >> (64 - ieee754::traits<float64_t>::mantissaBitCnt);
+    output.y = mantissa64 << (ieee754::traits<float64_t>::mantissaBitCnt);
+
+    return output;
+}
+
+NBL_CONSTEXPR_INLINE_FUNC uint64_t packFloat64(uint32_t zSign, int zExp, uint32_t zFrac0, uint32_t zFrac1)
+{
+    uint32_t2 z;
+
+    z.x = zSign + (uint32_t(zExp) << 20) + zFrac0;
+    z.y = zFrac1;
+
+    uint64_t output = 0u;
+    output |= (uint64_t(z.x) << 32) & 0xFFFFFFFF00000000ull;
+    output |= uint64_t(z.y);
+    return  output;
+}
```

Comment:
where do you use it, it seems to make no sense, you should have a function without two separate uint32_t for the mantissa parts